pkg/cli/find.go

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+*
+* Copyright 2018 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+package cli
+
+import (
+	"errors"
+
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/domains"
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/projects"
+)
+
+// FindDomain uses the user's input (name/UUID) to find a specific domain within the token scope.
+func FindDomain(userInput string) (*Domain, error) {
+	identityV3, _ := getServiceClients()
+	d := new(Domain)
+
+	// check if userInput is a UUID
+	tmpD, err := domains.Get(identityV3, userInput).Extract()
+	if err == nil {
+		d.ID = tmpD.ID
+		d.Name = tmpD.Name
+	} else {
+		// userInput appears to be a name so we do domain listing
+		// restricted to the name
+		page, err := domains.List(identityV3, domains.ListOpts{Name: userInput}).AllPages()
+		if err != nil {
+			return nil, err
+		}
+		dList, err := domains.ExtractDomains(page)
+		if err != nil {
+			return nil, err
+		}
+
+		// no need to iterate over the entire list, if there are multiple domains
+		if len(dList) > 1 {
+			return nil, errors.New("more than one domain exists with the name " + userInput)
+		}
+
+		for _, dInList := range dList {
+			d.ID = dInList.ID
+			d.Name = dInList.Name
+		}
+	}
+
+	return d, nil
```

Comment:
As far as I can see, you need another case here (and also in `FindProject`) for when the `userInput` does not refer to any existing domain, i.e. when the listing returns zero entries. Right now, you would be returning an uninitialized (i.e. all-zeroed) Domain in that case.